### PR TITLE
method for scrolling and searching

### PR DESCRIPTION
**Pull request by ashby-bowles**

### CODE REVIEW

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 1]
```diff
@@ -416,6 +416,60 @@ private Locomotive performSwipe(SwipeElementDirection direction, WebElement elem
...
+
+
+    public WebElement swipeTo(SwipeElementDirection s, By by, int attempts){
+
+        int i;
+
+        if (isPresentWait(by)){
+            return driver.findElement(by);
+        }
+        else {
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: `else` should be on the same line as `{` one line above.

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 2]
```diff
@@ -416,6 +416,60 @@ private Locomotive performSwipe(SwipeElementDirection direction, WebElement elem
         return this;
     }
 
+
+
+
+    public WebElement swipeTo(SwipeElementDirection s, By by, int attempts){
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Might be good to create a few overload methods for this
- `public WebElement swipeTo(By by)`
  - `SwipeDirection` could default to `UP` (scrolling down on the screen)
  - `attempts` has a default value of X (pick a number??)
- `public WebElement swipeTo(By by, SwipeDirection direction)`
- `public WebElement swipeTo(By by, int attempts)`

They all overload to this method.

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 3]
```diff
@@ -416,6 +416,60 @@ private Locomotive performSwipe(SwipeElementDirection direction, WebElement elem
...
+                swipeCenter(s);
+                if (isPresentWait(by)) {
+                    return driver.findElement(by);
+                }
+
+            }
+
+        }
+        System.err.println("WARN: Element" + by.toString() + "does not exist!");
+        return null;
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: I don't know how I feel about returning `null`, what would be the benefit to not failing the test here?

File: src/main/java/com/joss/conductor/mobile/Locomotive.java (hunk 2 above)
Comment (jossjacobo) [reply src/main/java/com/joss/conductor/mobile/Locomotive.java]: I forgot the overloads for the `String id`s

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 4]
```diff
@@ -416,6 +416,60 @@ private Locomotive performSwipe(SwipeElementDirection direction, WebElement elem
...
+
+
+    }
+
+
+    public WebElement swipeTo(SwipeElementDirection s, String id, int attempts){
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: This should overload to the above method, therefore the logic for swiping to an element is only in one place.

File: src/main/java/com/joss/conductor/mobile/Locomotive.java (hunk 1 above)
Comment (ashby-bowles) [reply src/main/java/com/joss/conductor/mobile/Locomotive.java]: done

File: src/main/java/com/joss/conductor/mobile/Locomotive.java (hunk 2 above)
Comment (ashby-bowles) [reply src/main/java/com/joss/conductor/mobile/Locomotive.java]: ah good thinking! done

File: src/main/java/com/joss/conductor/mobile/Locomotive.java (hunk 3 above)
Comment (ashby-bowles) [reply src/main/java/com/joss/conductor/mobile/Locomotive.java]: well my thinking was that returning *null* should fail the test pretty fast if a test tries to interact with it thinking that it is WebElement. I can add some more error logging maybe if that'd be helpful?

Comment (jossjacobo) [reply src/main/java/com/joss/conductor/mobile/Locomotive.java]: Hmm ok, I think it's fine.

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 5]
```diff
@@ -416,6 +416,53 @@ private Locomotive performSwipe(SwipeElementDirection direction, WebElement elem
         return this;
     }
 
+
+    public WebElement swipeTo(SwipeElementDirection s, By by, int attempts) {
+
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Too much vertical spacing, make it consistent with the rest of the code and we are good :)

**Pull request merged**